src/core/qgsgeometryvalidator.cpp

```diff
@@ -230,6 +230,28 @@ void QgsGeometryValidator::run()
         char res = GEOSisValidDetail_r( handle, g0.get(), GEOSVALID_ALLOW_SELFTOUCHING_RING_FORMING_HOLE, &r, &g1 );
         if ( res != 1 )
         {
+          static QgsStringMap translatedErrors;
+
+          if ( translatedErrors.empty() )
+          {
+            // Copied from https://git.osgeo.org/gitea/geos/geos/src/branch/master/src/operation/valid/TopologyValidationError.cpp
+            translatedErrors.insert( QStringLiteral( "Topology Validation Error" ), QObject::tr( "Topology Validation Error", "GEOS Error" ) );
+            translatedErrors.insert( QStringLiteral( "Repeated Point" ), QObject::tr( "Repeated Point", "GEOS Error" ) );
```

Comment:
Is that a title?  otherwise point has to be lowercase - same for self-intersections and duplicate rings and invalid coordinate.